csv/generic/shared/src/test/scala/fs2/data/csv/generic/CsvRowDecoderTest.scala

```diff
@@ -90,4 +97,446 @@ object CsvRowDecoderTest extends SimpleIOSuite {
+      Left(new DecoderError("unexpect end of row", None)),
+      Right(TestData("Bob Smith", "80", "description 2")),
+      Left(new DecoderError("unexpect end of row", None))
+    )
+
+    val stream = Stream
+      .emit(content)
+      .covary[Fallible]
+      .through(lenient.attemptDecodeSkippingHeaders[TestData](separator = ','))
+      .compile
+      .toList
+
+    stream match {
+      case Right(actual) => expect.eql(expected, actual)
+      case Left(x)       => failure(s"Stream failed with value $x")
+    }
+  }
+
+  pureTest("Parser should return all decoder results when using attemptDecodeWithoutHeaders") {
+    val content =
+      """John Doe,47,description 1
+        |Jane Doe,50
+        |Bob Smith,80,description 2
+        |Alice Grey,78
+        |""".stripMargin
+
+    val expected: List[Either[Throwable, TestData]] = List(
+      Right(TestData("John Doe", "47", "description 1")),
+      Left(new DecoderError("unexpect end of row", None)),
+      Right(TestData("Bob Smith", "80", "description 2")),
+      Left(new DecoderError("unexpect end of row", None))
+    )
+
+    val stream = Stream
+      .emit(content)
+      .covary[Fallible]
+      .through(lenient.attemptDecodeWithoutHeaders[TestData](separator = ','))
+      .compile
+      .toList
+
+    stream match {
+      case Right(actual) => expect.eql(expected, actual)
+      case Left(x)       => failure(s"Stream failed with value $x")
+    }
+  }
+
+  pureTest("decodeUsingHeaders should map data to case class matching the header name to the case class field name") {
+    implicit val decoder: CsvRowDecoder[Test, String] = testDecoder
+    implicit val testEq: Eq[Test] = Eq.fromUniversalEquals
+
+    val content =
+      """s,i,j
+        |a,12,3
+        |""".stripMargin
+
+    val expected: List[Test] = List(
+      Test(12, "a", Some(3))
+    )
+
+    val stream = Stream
+      .emit(content)
+      .covary[Fallible]
+      .through(decodeUsingHeaders[Test](','))
+      .compile
+      .toList
+
+    stream match {
+      case Right(actual) => expect.eql(expected, actual)
+      case Left(x)       => failure(s"Stream failed with value $x")
+    }
+  }
+
+  pureTest("decodeUsingHeaders should succeed if an optional field is missing") {
+    implicit val decoder: CsvRowDecoder[Test, String] = testDecoder
+    implicit val testEq: Eq[Test] = (a, b) => a == b
+
+    val content =
+      """s,i
+        |a,12
+        |""".stripMargin
+
+    val expected: List[Test] = List(
+      Test(12, "a", None)
+    )
+
+    val stream = Stream
+      .emit(content)
+      .covary[Fallible]
+      .through(decodeUsingHeaders[Test](','))
+      .compile
+      .toList
+
+    stream match {
+      case Right(actual) => expect.eql(expected, actual)
+      case Left(x)       => failure(s"Stream failed with value $x")
+    }
+  }
+
+  pureTest("decodeUsingHeaders should succeed if a required field with default value is missing") {
```

Comment:
This one should fail on Scala 3 due to the missing default support, so let's remove it (for now).